disaggregated_memory/boards/arm_main_board.py

```diff
@@ -0,0 +1,469 @@
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+# The goal of this board is to combine the gem5-only and the gem5-SSt boards
+# into one single board.
+import os
+import sys
+from abc import ABCMeta
+from typing import (
+    List,
+    Sequence,
+    Tuple,
+)
+
+# all the source files are one directory above.
+sys.path.append(
+    os.path.abspath(os.path.join(os.path.dirname(__file__), os.path.pardir))
+)
+
+from memories.external_remote_memory_v2 import ExternalRemoteMemoryV2
+
+import m5
+from m5.objects import (
+    AddrRange,
+    ArmSystem,
+    BadAddr,
+    ExternalMemory,
+    IOXBar,
+    NoncoherentXBar,
+    Port,
+    SrcClockDomain,
+    Terminal,
+    VncServer,
+    VoltageDomain,
+)
+from m5.objects.ArmFsWorkload import ArmFsLinux
+from m5.objects.ArmSystem import (
+    ArmDefaultRelease,
+    ArmRelease,
+)
+from m5.objects.RealView import (
+    VExpress_GEM5_Base,
+    VExpress_GEM5_Foundation,
+)
+from m5.util.fdthelper import (
+    Fdt,
+    FdtNode,
+    FdtProperty,
+    FdtPropertyStrings,
+    FdtPropertyWords,
+    FdtState,
+)
+
+from gem5.components.boards.arm_board import ArmBoard
+from gem5.components.cachehierarchies.abstract_cache_hierarchy import (
+    AbstractCacheHierarchy,
+)
+from gem5.components.memory.abstract_memory_system import AbstractMemorySystem
+from gem5.components.processors.abstract_processor import AbstractProcessor
+from gem5.utils.override import overrides
+
+
+class ArmComposableMemoryBoard(ArmBoard):
+    """
+    A high-level ARM board that can zNUMA-capable systems with a remote
+    memories. This board is extended from the ArmBoard from Gem5 standard
+    library. This board assumes that you will be booting Linux. This board can
+    be used to do disaggregated ARM system research while accelerating the
+    simulation using kvm.
+
+    The revised ArmComposableMemoryBoard combines the older boards into one
+    single board to make the boards compatible with both gem5 and SST.
+
+    **Limitations**
+    * kvm is only supported in a gem5-only setup.
+
+    @params
+    TODO
+    """
+
+    __metaclass__ = ABCMeta
```

Comment:
Why is this here?

Reply:
I'll fix the TODO. I copy-pasted the `__metaclass__`. I'll get rid of that.

Reply:
Done!